### DISCUSSION TITLE
List of used columns/measures in the Power BI report

### DISCUSSION THREAD

**Discussion by arya4it** [DaxStudio/DaxStudio#875] [Q&A] [answered]
Hi,
We have a lot of Columns and Measure in the report.
Is there a way to get the list of **used** columns/measures in the Power BI report using DAX Studio.
Thanks
Arya

**Comment by dgosbell** [comment, marked as answer]
DAX Studio only connects to the data model. It does not know anything about the report or visuals. 

This has been asked before here https://github.com/DaxStudio/DaxStudio/discussions/808  there are tools like the one listed in the previous link, but they only scan the current report, but you could have columns used by other pbix reports, paginated reports or things like Analyze in Excel. So it is very hard to say if a given column/measure is used or not.

**Comment by bcobrien** [comment]
I am confused on this so can you get a list of all the fields by tables using Dax Studio? I inherited some massive files that need to be cleaned up.  Measures are one thing but it feels like there must be a way to get an inventory of all the columns.
I appreciate everyone's help.

**Comment by marcosqlbi** [comment]
@bcobrien You cannot be 100% sure of the columns used in case you publish the dataset on Power BI service and then someone creates a report using a published dataset. If your only concern is the report in the same PBIX file, you can use the Power BI Cleaner tool: https://www.thebiccountant.com/tools-power-bi-cleaner/
